### PR TITLE
[upgrade] Upgrade Zephyr to unified kernel

### PR DESCRIPTION
**Pull request by jprestwo**
Signed-off-by: James Prestwood <james.prestwood@intel.com>

### CODE REVIEW

**Comment by jprestwo** [comment]
If you have verified a module as working please comment here to let me know. So far this is what I have:

- Timers: working (tested samples/Timers.js)
- Events: working (tested samples/tests/Events.js)
- GPIO: tentatively working (tested samples/arduino/digital/Button.js)
- PWM: tentatively working (tested a singled PWM output to blink an LED)
- I2C: not working (@brianjjones verified)
- BLE: not working (@jimmy-huang verified)
- UART: works on QEMU, not working on A101

**Comment by sergioamr** [comment]
AShell: is now working. Will have to test more deeply next week.

**Comment by grgustaf** [comment]
PWM.js is not working. Only the IO3 pin seems to behave correctly.

**Comment by pfalcon** [comment]
Can you please share plans for this patch? Now that 0.1 has been tagged, any chance that this can be merged? I understand there're some regressions, but there's no other way than to upgrade, 1.6.0 is going to be released in few days. I'd say it's better to get everyone on the same base and pull together to resolve regressions. (As an example, I already can't use the revision of Zephyr which Z.js uses so far - my (and Linaro's) interest is to support more boards for Z.js, and Zephyr revision used by Z.js simply doesn't have those boards). Thanks.

**Comment by grgustaf** [comment]
We are working to merge it ASAP. We were close last night and there is one more thing @jprestwo's been asked to solve today, re: running out of memory on the WebBluetoothGroveLcdDemo.js.

File: scripts/analyze.sh [hunk 1]
```diff
@@ -103,26 +94,20 @@ if [ $? -eq 0 ]; then
     >&2 echo Using module: PWM
     MODULES+=" -DBUILD_MODULE_PWM"
     echo "CONFIG_PWM=y" >> prj.conf.tmp
-    echo "CONFIG_PWM_QMSI_NUM_PORTS=4" >> prj.conf.tmp
```
Comment (grgustaf) [inline scripts/analyze.sh]: Don't remove this line. It still works, and this is still required to make the A101 PWM pins work (apart from the first one).

**Comment by jimmy-huang** [comment]
lgtm, aside from the fact that there's a known issue with small memory allocation that would fail on the ble demo, and @jprestwo is going to fix the pool allocation.

**Comment by grgustaf** [comment]
I've rebased this patchset [here](https://github.com/grgustaf/zephyr.js-1/tree/pr-450-rebase) but it results in the identical source. However, I noticed that [Jimmy's IPM patch](https://github.com/01org/zephyr.js/pull/450/commits/bb1fef9a4333211847e7a7a8547bd530de9f820a) updates the iotivity-constrained submodule, which he didn't intend to do. So I don't know if he accidentally did that or @jprestwo did it while rebasing or something. Want to run that by @jprestwo before we merge.

**Comment by poussa** [comment]
If we don't have a good solution for BLE demo today, we should merge this and work on the demo fix after that.

**Comment by jimmy-huang** [comment]
@jprestwo confirmed it is ok, so I am going to merge this.

**Pull request merged**